src/proxy/outbound.rs

```diff
@@ -46,25 +47,32 @@ impl Outbound {
     pub(super) async fn run(self) {
         let addr = self.listener.local_addr().unwrap();
         info!("outbound listener established {}", addr);
-
         let accept = async move {
             loop {
                 // Asynchronously wait for an inbound socket.
                 let socket = self.listener.accept().await;
+                let start_outbound_instant = Instant::now();
                 match socket {
                     Ok((stream, remote)) => {
                         info!("accepted outbound connection from {}", remote);
                         let cfg = self.cfg.clone();
-                        let oc = OutboundConnection {
+                        let mut oc = OutboundConnection {
                             cert_manager: self.cert_manager.clone(),
                             workloads: self.workloads.clone(),
                             cfg,
                         };
                         tokio::spawn(async move {
                             let res = oc.proxy(stream).await;
                             match res {
-                                Ok(_) => info!("outbound proxy complete"),
-                                Err(ref e) => warn!("outbound proxy failed: {}", e),
+                                Ok(_) => info!(
+                                    "outbound proxy complete ({}ms)",
+                                    start_outbound_instant.elapsed().as_millis()
```

Comment:
can we add a format that is dynamic like Go's duration format? we don't want to use ms to log a connection that took 10hrs